### ISSUE TITLE
Offering help: Accessibility/legibility of graphics

### ISSUE THREAD

**Issue by Cottser** [ColemakMods/mod-dh#19]
Hello! I'm trying to use one of the layout graphics (from https://colemakmods.github.io/mod-dh/keyboards.html but this issue applies to all layout graphics) as an on-screen reference while I'm learning Colemak Mod-DH.

I'm not having the best time using the graphics for this purpose, and before I get into the details I'd like to let you know that I'm proposing a solution and offering to help implement it in the form of a pull request :)

So I was going through setting up https://colemakmods.github.io/mod-dh/gfx/mod_dh_keyb_matrix.png as an on-screen reference. This made me realize that the layout graphics used on the site could really use better contrast. My only vision problem is a medium level of nearsightedness that I wear glasses for and it doesn't take much of a squint to make the keys illegible for me. So I'd wager that these graphics are currently very difficult for folks with certain vision problems to read.

I would suggest changing the lettering to white and darkening the background colours to keep with the current look and feel while improving the legibility of the layout graphics. The alternative is to go pastel which wouldn't mesh with the rest of the site design IMO.

If you provide source files for the graphics, I'd be happy to produce higher-contrast versions and create a pull request. Or if you'd prefer to do it yourself I can recommend the following tool to check against WCAG guidelines for colour contrast on the web: https://webaim.org/resources/contrastchecker/

Thanks for your time.

**Comment by stevep99** [comment]
Hi, thanks for your feedback.

I think you make a valid point, I'm not surprised there is scope for improvement with the images.

The graphics are all generated using an app I have written for the express purpose of visualizing keyboard layouts. So rather than changing the images as a one-off, I'd prefer to make changes to the code that generates them. I had been planning on putting this code on GitHub at some point, but it's still on my to-do list. In the meantime, I'll try out some changes, and then perhaps get your feedback?

**Comment by Cottser** [comment]
Sounds good, that’s nice that they are generated!

If it would help you I can provide suggestions in the form of hex colour codes, but otherwise I’m more than happy to provide feedback. Thanks for your time and consideration.

**Comment by stevep99** [comment]
I generated this [test image](https://colemakmods.github.io/mod-dh/experimental/gfx/mod_dh_keyb_matrix.png) with white lettering with black outline to help contrast, what do you think?

**Comment by Cottser** [comment]
At a glance (and squint) it does look like a big improvement.

With those changes, the blue and purple backgrounds have sufficient contrast and the pink just needs a slight darkening. The teal and green colours could use a more significant darkening. I can provide hex colours (or RGB or other) if helpful. Thanks!

**Comment by DreymaR** [comment]
To me, the test image looks evil and annoying. But maybe that's a necessary sacrifice...?

**Comment by Cottser** [comment]
Could you please elaborate?

**Comment by DreymaR** [comment]
There is a confusing false 3D effect going on with the added edges, together with optical illusion effects including the Grid Illusion that make the image very uneasy on my eyes. It bulges and pops, making me a bit seasick. Maybe a darker background would work better, I don't know.

**Comment by stevep99** [comment]
What do you mean by "false 3D" effect? Do you mean the shadows under the keys? The existing images also have those. Or do you mean the black border around the white letters? 

I'm also not totally keen on the white lettering for some reason, I considered that option originally but to my eyes the black letters looked better - but then I'm not a expert on these things.

**Comment by DreymaR** [comment]
The shadows are for an intended 3D effect. A side effect of the high contrast to the white letters is that they seemingly pop up but then they don't, and with that added to the grid illusion the net result is too screeching for me. Not sure I can explain it better. If you don't see it, maybe it's just me...?

**Comment by stevep99** [comment]
Perhaps I'm not getting quite the same effect as you, but there is something I don't like about the white lettering version, but I wasn't quite sure why, so maybe you are onto something.

For comparison I also did a new version with [black text with white border](https://github.com/ColemakMods/mod-dh/blob/gh-pages/experimental/gfx/mod_dh_keyb_matrix_1.png), it's similar to the existing but the text is perhaps more bold and clearer?

**Comment by Cottser** [comment]
@DreymaR thanks for explaining!

@stevep99 I can get behind the new version, definitely easier to read and would be a good improvement over the current. The thicker and darker letters seem to be doing most of the work, I'm not sure that the thicker white border/shadow/glow is doing much but tough to say without seeing it.

**Comment by stevep99** [comment]
OK, after going back and forth, I'm [come up with this new graphic](https://github.com/ColemakMods/mod-dh/blob/gh-pages/experimental/gfx/mod_dh_keyb_matrix_2.png) which keeps the original grey lettering, but bold, and with a sharp black border for definition. I think this is probably the best candidate yet, but opinions are welcome.

**Comment by Cottser** [comment]
@stevep99 the thicker letters definitely help - though it looks like the dark glow/shadow may be hurting at this point by making the outlines of the letters less distinct.

**Comment by stevep99** [comment]
I finally got around to deciding what to do with these images. I thought the fully black bold version was overkill, wasn't keen on the white on either. But then I figured out I could do semi-bold as a compromise. Combined with making the text darker, this makes the text clearer without making the graphics radically different. Images now updated.

**Comment by Cottser** [comment]
Delayed response but thanks, this is certainly an improvement.
